src/bug-filing/common/issue-filing-controller-impl.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserAdapter } from '../../background/browser-adapter';
+import { BaseStore } from '../../common/base-store';
+import { EnvironmentInfo } from '../../common/environment-info-provider';
+import { FileIssueClickService } from '../../common/telemetry-events';
+import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { BugServiceProperties, UserConfigurationStoreData } from '../../common/types/store-data/user-configuration-store';
+import { BugFilingServiceProvider } from '../bug-filing-service-provider';
+
+export type IssueFilingController = {
+    fileIssue: (serviceKey: FileIssueClickService, issueData: CreateIssueDetailsTextData) => void;
```

Comment:
`FileIssueClickService` was defined in telemetry-events.ts as the type for the `service` in `FileIssueClickTelemetryData`, but now it's being used more broadly. I think Telemetry code's parameter type should depend on the File Issue code's service type, not the other way around. Probably also rename it to `IssueFilingServiceKey` as `FileIssueClickService` is confusing (to me, at least) in this context. Oh, and we should add `azureBoards` as a possible value.

Reply:
I was going to clean up that type on a different PR, but since it surface here...
The type is not giving us any real advantage. Having a test to ensure uniqueness of the key across the services is enough. Base on this, I remove the type and use `string` instead.